src/host_vm.rs

```diff
@@ -282,11 +282,12 @@ impl<T: GuestStagePagingMode> HostVmLoader<T> {
             }
         }
 
-        // Host guarantees that the host pages it returns start at T::TOP_LEVEL_ALIGN-aligned block,
-        // and because we built the HwMemMap with a minimum region alignment of T::TOP_LEVEL_ALIGN
+        // Host guarantees that the host pages it returns start at 2M-aligned block,
+        // and because we built the HwMemMap with a minimum region alignment of 2M,
         // any discontiguous ranges are also guaranteed to be aligned.
         //
         // Now fill in the address space, inserting zero pages around the kernel/initramfs/FDT.
+        assert!(self.guest_ram_base.is_aligned(PageSize::Size2M));
```

Comment:
@abrestic-rivos looking at the code that comes after this, I'm wondering if we should ensure that `num_pages`, `num_kernel_pages`, etc are covering a 2M size multiple. Because if that's not the case, we might end up with `current_gpa` not being aligned on 2M, and that would mean some `zero_ranges` range might not be properly aligned. What do you think?

Reply:
yes, the kernel & initramfs regions should be 2M-aligned and rounded up to 2M in length.